lib/usePantry.ts

```diff
@@ -90,33 +90,75 @@ const getGitDistribution = ({ pkg, url: urlstr, ref }: { pkg: Package, url: stri
 }
 
 const getDistributable = async (pkg: Package) => {
-  const moustaches = useMoustaches()
-
-  const yml = await hooks.usePantry().project(pkg).yaml()
-
-  if (yml.distributable?.git) {
-    console.warn("brewkit: using distributable.git instead of distributable.url is deprecated")
-    return getGitDistribution({ pkg, ...yml.distributable})
-  }
-  if (yml.distributable?.url?.startsWith("git")) {
-    return getGitDistribution({ pkg, ...yml.distributable})
-  }
-
-  let urlstr = getRawDistributableURL(yml)
-  if (!urlstr) return
-  let stripComponents: number | undefined
-  if (isPlainObject(yml.distributable)) {
-    stripComponents = flatmap(yml.distributable["strip-components"], coerceNumber)
+  const moustaches = useMoustaches();
+
+  const yml = await hooks.usePantry().project(pkg).yaml();
+  const dists = isArray(yml.distributable) ? yml.distributable : [yml.distributable]
+  for (const dist of dists) {
+    if (!dist) continue;
+    //FIXME: Add check for Git dists as well
+    if (dist.git) {
+      console.warn(
+        "brewkit: using distributable.git instead of distributable.url is deprecated",
+      );
+      return getGitDistribution({ pkg, ...dist });
+    }
+
+    if (dist.url?.startsWith("git")) {
+      return getGitDistribution({ pkg, ...dist });
+    }
+
+    let urlstr = getRawDistributableURL(dist);
+    if (!urlstr) continue;
+
+    let v = pkg.version
+    const stripComponents = flatmap(dist["strip-components"], coerceNumber)
+
+    if (isPlainObject(dist)) {
+      if (dist.rewrite?.match) {
+        const v_raw = v.raw.replace(
+          new RegExp(dist.rewrite["match"], "gi"),
+          dist.rewrite["with"],
+        );
+        v = {
+        raw: v_raw,
+        major: pkg.version.major,
+        minor: pkg.version.minor,
+        patch: pkg.version.patch,
+        components: pkg.version.components,
+        prerelease: pkg.version.prerelease,
+        build: pkg.version.build,
+        eq: pkg.version.eq,
+        neq: pkg.version.neq,
+        gt: pkg.version.gt,
+        gte: pkg.version.gte,
+        lt: pkg.version.lt,
+        lte: pkg.version.lte,
+        compare: pkg.version.compare,
+      }
+      }
```

Comment:
this would be simpler as:
```typescript
        const raw = v.raw.replace(
          new RegExp(dist.rewrite["match"], "gi"),
          dist.rewrite["with"],
        );
        v = { ...v, raw }
```

Reply:
That wouldn't work, `SemVer` is class, not an object. It can't be spread (I think I tried it and Deno yelled at me)